### ISSUE TITLE
C files not vendored

### ISSUE THREAD

**Issue by iand** [kardianos/govendor#124]
I'm using govendor on a project that uses github.com/go-gl/glfw/v3.1/glfw which contains some C bindings and header files in a subdirectory. After using `govendor add +external` these files were not copied into the vendor directory of my project and so it fails to compile:

```
vendor/github.com/go-gl/glfw/v3.1/glfw/context.go:4:37: fatal error: glfw/include/GLFW/glfw3.h: No such file or directory
 //#include "glfw/include/GLFW/glfw3.h"
```

Manually copying the C files into the right place in the vendor directory fixes the problem.

**Comment by iand** [comment]
Some additional info. Using govendor fetch does not fetch the subdirectory containing the C files even though it does update the Go files in the vendored package.
I tried both

`govendor fetch github.com/go-gl/glfw/v3.1/glfw`

and

`govendor fetch github.com/go-gl/glfw/v3.1/glfw/...`

**Comment by kardianos** [comment]
Try running 
`govendor remove github.com/go-gl/glfw/v3.1/glfw/... && govendor fetch github.com/go-gl/glfw/v3.1/glfw/^`

This will copy the entire directory tree.

**Comment by iand** [comment]
Thanks for the speedy response. I get the following error when running the second part of that command (I ran the remove successfully):

`Error: Cannot have a sub-tree "github.com/go-gl/glfw/v3.1/glfw" contain sub-packages ["github.com/go-gl/glfw/v3.1/glfw^"]`

**Comment by kardianos** [comment]
I'm having trouble reproducing with HEAD. Can you post the result of `govendor list github.com/go-gl/glfw/...`

**Comment by kardianos** [comment]
Do you still have files in the folder when you added them manually?
what if you ran `rm -rf vendor/github.com/go-gl/glfw` then 
`govendor fetch github.com/go-gl/glfw/v3.1/glfw/^`

**Comment by iand** [comment]
Here's the result of `govendor list github.com/go-gl/glfw/...` executed immediately after getting the error about sub-tree/sub-packages:

```
 e  github.com/go-gl/glfw/v3.1/glfw     
  m github.com/go-gl/glfw/v3.1/glfw^    
```

**Comment by iand** [comment]
Some progress... I deleted everything under the vendor directory and re-ran the `govendor fetch github.com/go-gl/glfw/v3.1/glfw/^` command. That completed successfully and vendored all the C files as well. 

Here's the relevant portion of `govendor list` showing the other vendored packages I need:

```
 v  github.com/go-gl/glfw/v3.1/glfw              
 e  github.com/go-gl/gl/v4.1-core/gl             
 e  github.com/go-gl/mathgl/mgl32                
 e  golang.org/x/image/math/f32                  
```

Do i just need to use the sub-folder package specifier for vendoring packages with associated C like this or should `govendor add +external` handle those?

As an experiment I deleted some of the vendored files and verified that `govendoe

**Comment by iand** [comment]
Now the sub-folder problem re-appears when I try to vendor another package alongside glfw:

```
$ govendor fetch github.com/go-gl/gl/v4.1-core/gl/^
Error: Cannot have a sub-tree "github.com/go-gl/gl/v4.1-core/gl" contain sub-packages ["github.com/go-gl/gl/v4.1-core/gl^"]
```

So, to be clear these are the commands I used from the root of my package:

```
$ rm -rf vendor/*
$ govendor fetch github.com/go-gl/glfw/v3.1/glfw/^
$ govendor fetch github.com/go-gl/gl/v4.1-core/gl/^
```

which then gives the error above.

**Comment by kardianos** [comment]
I'm sorry, I'm still not able to reproduce. If you are willing to continue, could you execute the following commands and send me the output?
```bash
go get -u github.com/kardianos/govendor
cd $PROJECT_ROOT
go env
rm -rf vendor/*
govendor list -v
govendor fetch github.com/go-gl/glfw/v3.1/glfw/^
govendor list -v
cat vendor/vendor.json
govendor fetch github.com/go-gl/gl/v4.1-core/gl/^
```

Thank you for your patients in helping understand this.

**Comment by iand** [comment]
I ran those commands and... no errors. I then used the `govendor fetch` to vendor the remaining 2 dependencies and that worked fine too. I am at a loss to explain why because I ran `go get -u github.com/kardianos/govendor` just before I started using the tool to ensure I had the latest version.

I'm happy to use this as a workaround for `govendor add +external` not copying the C files. Thanks for all your help.
